From 630afe8318340f4e925f72e54c0bd54c97bfb53e Mon Sep 17 00:00:00 2001
From: ferhung-mtk <ferhung27@gmail.com>
Date: Sun, 17 Apr 2016 17:03:36 +0800
Subject: [PATCH 02/10] MediaTek: setting selinux to permissive.

Signed-off-by: sub77 <sub77@ymail.com>
---
 init/init.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index b332588..2171066 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -917,6 +917,8 @@ static bool selinux_is_disabled(void)
 
 static bool selinux_is_enforcing(void)
 {
+    return false;  /*return false then set to permissive*/
+
     if (ALLOW_DISABLE_SELINUX) {
         return selinux_status_from_cmdline() == SELINUX_ENFORCING;
     }
-- 
2.9.3

