From eb7e58296e076e5c75488c27198dde75d7461fbd Mon Sep 17 00:00:00 2001
From: ferhung-mtk <ferhung27@gmail.com>
Date: Sun, 17 Apr 2016 17:09:53 +0800
Subject: [PATCH 10/10] MediaTek: fix offline charging mod.

Signed-off-by: sub77 <sub77@ymail.com>
---
 init/init.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/init.cpp b/init/init.cpp
index 4b5d794..6f54282 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -1055,7 +1055,7 @@ static int charging_mode_booting(void) {
         return 0;
 
     close(f);
-    return ('1' == cmb);
+    return ('8' == cmb);
 #endif
 }
 
-- 
2.9.3

