From 167b6defb01686e22fede1d5471421a40916cce6 Mon Sep 17 00:00:00 2001
From: ferhung-mtk <ferhung27@gmail.com>
Date: Sun, 3 Apr 2016 10:33:45 +0800
Subject: [PATCH 2/2] MediaTek: fix reboot for MTK.

Signed-off-by: sub77 <sub77@ymail.com>
---
 twrp-functions.cpp | 33 +++++++++++++++++----------------
 twrp.cpp           |  6 +++++-
 2 files changed, 22 insertions(+), 17 deletions(-)

diff --git a/twrp-functions.cpp b/twrp-functions.cpp
index 8cc094f..611eba6 100644
--- a/twrp-functions.cpp
+++ b/twrp-functions.cpp
@@ -45,12 +45,14 @@
 #include "variables.h"
 #include "bootloader.h"
 #include "cutils/properties.h"
+#ifdef ANDROID_RB_POWEROFF
 #include "cutils/android_reboot.h"
-#include <sys/reboot.h>
+#endif
 #endif // ndef BUILD_TWRPTAR_MAIN
 #ifndef TW_EXCLUDE_ENCRYPTED_BACKUPS
 	#include "openaes/inc/oaes_lib.h"
 #endif
+#include "cutils/android_reboot.h"
 
 extern "C" {
 	#include "libcrecovery/common.h"
@@ -537,46 +539,45 @@ int TWFunc::tw_reboot(RebootCommand command)
 	switch (command) {
 		case rb_current:
 		case rb_system:
+			Update_Log_File();
 			Update_Intent_File("s");
 			sync();
 			check_and_run_script("/sbin/rebootsystem.sh", "reboot system");
-#ifdef ANDROID_RB_PROPERTY
-			return property_set(ANDROID_RB_PROPERTY, "reboot,");
-#elif defined(ANDROID_RB_RESTART)
-			return android_reboot(ANDROID_RB_RESTART, 0, 0);
-#else
 			return reboot(RB_AUTOBOOT);
-#endif
 		case rb_recovery:
 			check_and_run_script("/sbin/rebootrecovery.sh", "reboot recovery");
 #ifdef ANDROID_RB_PROPERTY
-			return property_set(ANDROID_RB_PROPERTY, "reboot,recovery");
+			property_set(ANDROID_RB_PROPERTY, "reboot,recovery");
 #else
 			return __reboot(LINUX_REBOOT_MAGIC1, LINUX_REBOOT_MAGIC2, LINUX_REBOOT_CMD_RESTART2, (void*) "recovery");
 #endif
+			sleep(5);
+			return 0;
 		case rb_bootloader:
 			check_and_run_script("/sbin/rebootbootloader.sh", "reboot bootloader");
 #ifdef ANDROID_RB_PROPERTY
-			return property_set(ANDROID_RB_PROPERTY, "reboot,bootloader");
+			property_set(ANDROID_RB_PROPERTY, "reboot,bootloader");
 #else
 			return __reboot(LINUX_REBOOT_MAGIC1, LINUX_REBOOT_MAGIC2, LINUX_REBOOT_CMD_RESTART2, (void*) "bootloader");
 #endif
+			sleep(5);
+			return 0;
 		case rb_poweroff:
 			check_and_run_script("/sbin/poweroff.sh", "power off");
-#ifdef ANDROID_RB_PROPERTY
-			return property_set(ANDROID_RB_PROPERTY, "shutdown,");
-#elif defined(ANDROID_RB_POWEROFF)
-			return android_reboot(ANDROID_RB_POWEROFF, 0, 0);
-#else
-			return reboot(RB_POWER_OFF);
+#ifdef ANDROID_RB_POWEROFF
+			android_reboot(ANDROID_RB_POWEROFF, 0, 0);
 #endif
+			return reboot(RB_POWER_OFF);
+
 		case rb_download:
 			check_and_run_script("/sbin/rebootdownload.sh", "reboot download");
 #ifdef ANDROID_RB_PROPERTY
-			return property_set(ANDROID_RB_PROPERTY, "reboot,download");
+			property_set(ANDROID_RB_PROPERTY, "reboot,download");
 #else
 			return __reboot(LINUX_REBOOT_MAGIC1, LINUX_REBOOT_MAGIC2, LINUX_REBOOT_CMD_RESTART2, (void*) "download");
 #endif
+			sleep(5);
+			return 0;
 		default:
 			return -1;
 	}
diff --git a/twrp.cpp b/twrp.cpp
index 5d64694..0d29ef8 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -402,6 +402,10 @@ int main(int argc, char **argv) {
 		TWFunc::tw_reboot(rb_download);
 	else
 		TWFunc::tw_reboot(rb_system);
-
+#ifdef ANDROID_RB_RESTART
+	android_reboot(ANDROID_RB_RESTART, 0, 0);
+#else
+	reboot(RB_AUTOBOOT);
+#endif
 	return 0;
 }
-- 
2.9.3

